docs/rules/regex-shorthand.md

```diff
@@ -25,3 +25,17 @@ const regex = /\W/;
 const regex = /\W/i;
 const regex = /\d\.\w\-\D/i;
 ```
+
+
+## Options
+
+You can set the option like this:
+
+```js
+"unicorn/regex-shorthand": ["error", [
+  'charClassToMeta', // [0-9] -> [\d]
+  'charClassToSingleChar' // [\d] -> \d
```

Comment:
It's generally better for ESLint rules to be objects with true/false values, as it makes it easier to override. Like `{charClassToMeta: false}`.